cmd/kubeadm/app/preflight/checks.go

```diff
@@ -814,34 +817,34 @@ func (evc ExternalEtcdVersionCheck) getHTTPClient(config *tls.Config) *http.Clie
 		})
 		return &http.Client{
 			Transport: transport,
-			Timeout:   externalEtcdRequestTimeout,
+			Timeout:   evc.requestTimeout,
 		}
 	}
-	return &http.Client{Timeout: externalEtcdRequestTimeout, Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
+	return &http.Client{Timeout: evc.requestTimeout, Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
 }
 
-func getEtcdVersionResponse(client *http.Client, url string, target interface{}) error {
-	loopCount := externalEtcdRequestRetries + 1
+func (evc ExternalEtcdVersionCheck) getEtcdVersionResponse(client *http.Client, url string, target interface{}) error {
+	loopCount := evc.requestRetries + 1
 	var err error
 	var stopRetry bool
 	for loopCount > 0 {
-		if loopCount <= externalEtcdRequestRetries {
-			time.Sleep(externalEtcdRequestInterval)
+		if loopCount <= evc.requestRetries {
+			time.Sleep(evc.requestInterval)
 		}
 		stopRetry, err = func() (stopRetry bool, err error) {
 			r, err := client.Get(url)
 			if err != nil {
 				loopCount--
-				return false, nil
+				return false, err
```

Comment:
Right. L835 and L841 are the only two places we should change.

Reply:
Meaning the PR should only touch these two lines? It is currently a bit more involved due to the added unit tests, and trying to avoid the tests running for longer than they need by being able to specify the retry parameters. 

Is there a better approach to this? Thanks!

cc @timothysc